frontend/src/pages/Register/Register.js

```diff
@@ -0,0 +1,156 @@
+      width: '30%'
+    }
+  },
+  title: {
+    margin: theme.spacing(2),
+    textAlign: 'center',
+    fontWeight: 'bolder',
+    textTransform: 'uppercase'
+  }
+}));
+
+
+const Register = (props) => {
+  const classes = useStyles();
+  const history = useHistory();
+  const [state, setState] = React.useState({
+    email: '',
+    password: '',
+    passwordConfirmation: '',
+    snack: {
+      message: '',
+      open: false
+    },
+    errors: {
+      email: false,
+      password: false,
+      passwordConfirmation: false
+    },
+    form_disabled: false
+  });
+
+  // Updates given state with given value 
+  const handleChange = (prop) => (event) => {
+    setState({ ...state, [prop]: event.target.value });
+  };
+
+  // Handles submission of the form (button click)
+  // Validates form data for completion/length, making a backend request to Users DB for secret key autentication + 
+  // email uniqueness. If register succeeds, user is redirected to admin page and added to user DB. Otherwise, an error message appears. 
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    setState({ ...state, form_disabled: true });
+
+    const submission = {
+      username: state.email,
+      password: state.password,
+    };
+
+     // check password length
+     if (submission.password.length < 6) {
+      document.body.style.cursor= null;
+      setState({...state, errors: {email: false, password: true, passwordConfirmation: false}, form_disabled: false,  snack: {message: 'Password must be at least 6 characters long.', open: true}});
+      return;
+    }
+
+    // check passwords match
+    if (state.password !== state.passwordConfirmation) {
+      document.body.style.cursor= null;
+      setState({...state,errors: {email: false, password: true, passwordConfirmation: true}, form_disabled: false, snack: {message: 'Passwords Do Not Match.', open: true}});
+      return;
+    }
+
+    //Attempt to register with given credentials 
+    try {
+      const response = await fetch(`${config.backend.uri}/register`, {
+        method: 'POST',
+        headers: { 'Content-Type': 'application/json' },
+        body: JSON.stringify(submission)
+      });
+      //Successful Registration
+      if (response.ok) {
+        console.log("account pending approval");
+        history.push("/");
+      }
+      else if (response.status === 403) {
+        document.body.style.cursor= null;
+        setState({...state, form_disabled: false, errors: {email: true, password: false, passwordConfirmation: false}, snack: {message: 'Could not register account: Email already in use!', open: true}});
+      }
+    }
+    catch (error) {
+      document.body.style.cursor= null;
+      setState({...state, errors: {email: false, password: false, passwordConfirmation: false}, form_disabled: false, snack: {message: `An error occurred: ${error.message}`, open: true}});
+    }
+  };
+
+  const handleSnackClose = (event, reason) => {
```

Comment:
For this file and Login.js can you include some explanation on this handleSnackClose and why clickaway has a different behavior than everything else